pkg/storaged/vdos-panel.jsx

```diff
@@ -24,16 +24,25 @@ import { SidePanelRow } from "./side-panel.jsx";
 import {
     fmt_size, decode_filename,
     get_available_spaces, prepare_available_spaces,
-    get_config
 } from "./utils.js";
 import { dialog_open, TextInput, SelectSpace, SizeSlider, CheckBoxes } from "./dialog.jsx";
 
 const _ = cockpit.gettext;
 
+function vdo_package_name() {
```

Comment:
I am not a fan of moving this logic into the code itself... can we have a `quirks` file that can be controlled easily by the packaging scripts (spec and debian/rules)?

Reply:
I'm not much of a fan of it myself, but having it in the spec is not exactly great either. For quirks we could perhaps have some middle ground -- one *can* put quirks into the manifest, but if there are none the page would look at the OS and auto-detect features? So that we can distribute a manifest that works everywhere, without having to build and carefully configure?

Reply:
Isn' the spec file the exact right place for this?  That's where we know exactly whether whether vdo support is available what the package is called, no?

Reply:
> For quirks we could perhaps have some middle ground -- one can put quirks into the manifest, but if there are none the page would look at the OS and auto-detect features? 

I would put the quirks in their own file, completely separate from the manifest.  It could contain values per OS, and we could distribute a reasonable default version of it.  Something like this:

```
{
  "rhel": {
    "vdo_package": "vdo"
  },
  "centos": {
    "vdo_package": "vdo"
  }
}
```

Reply:
(But of course, YAGNI and all.  Just hard coding what we support into the code is a good start until we really find the need to be more flexible.)

Reply:
Such a distro → config map seems ok and rather generic; we could e.g. use a similar approach for the appstream metadata packages as well. But IMHO it's overkill to invent a second parallel hierarchy of data files and an API to access it from pages -- we have all that with manifests only. My concern is to fix manifests to be *portable*, not making them as small as possible.

I disagree that the spec file is the right place for that -- vdo may appear on Fedora or someone may install some COPR, or vdo may disappear from CentOS stream or CentOS 9, etc.

I much prefer runtime detection over build/configure time, but apparently we need to agree to disagree here.

So I see two ways, either of which I'd support:

 - Drop the os-release test in favor of actually asking the packaging system "do you know about the vdo package"? On our "blank /var/cache" test images that always incurs a 30 second package metadata download, but real-life systems usually have that cache, and thus it may not actually be so bad.

- Put this back into the manifest, but in a per-distro map.

WDYT?

Reply:
FTR, I'm currently leaning towards the second option (per-distro map in manifest), which is fast and efficient, and "quirkable".

Reply:
> I much prefer runtime detection over build/configure time, but apparently we need to agree to disagree here.

No, we agree.  Run-time detection is better, but not reliably possible here, as far as I can see. I am just looking for a way to change the value of "vdo_package" without having to change the actual code, which I am afraid might be to painful for those who actually want to change it.

Reply:
> FTR, I'm currently leaning towards the second option (per-distro map in manifest), which is fast and efficient, and "quirkable".

Yes, that's a good idea. We can tell people to install a override.json file if they ever need to tweak this.

Reply:
So what about keeping the `get_config` API and just changing it's implementation to work with the distro/config map?

Reply:
@mvollmer: That's what I'm doing right now. Nice!